Tests/NIOTests/SocketOptionChannelTest.swift

```diff
@@ -0,0 +1,129 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import XCTest
+
+final class SocketOptionChannelTest: XCTestCase {
+    var group: MultiThreadedEventLoopGroup!
+    var serverChannel: Channel!
+    var clientChannel: Channel!
+
+    struct CastError: Error { }
+
+    private func convertedChannel(file: StaticString = #file, line: UInt = #line) throws -> SocketOptionChannel {
+        guard let socketOptionChannel = self.clientChannel as? SocketOptionChannel else {
+            XCTFail("Unable to cast \(String(describing: self.clientChannel)) to SocketOptionChannel", file: file, line: line)
+            throw CastError()
+        }
+        return socketOptionChannel
+    }
+
+    override func setUp() {
+        self.group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        self.serverChannel = try? assertNoThrowWithValue(ServerBootstrap(group: group).bind(host: "127.0.0.1", port: 0).wait())
```

Comment:
isn't `setUp` `throws`? If yes, then I think it'd be better to do `try` instead of `try?`

Reply:
Nop, `setUp` is not throws.

Reply:
cool, please disregard then :)